spec/form_models/application_details_form_spec.rb

```diff
@@ -17,6 +17,98 @@
           .to eq('is not a valid URL')
       end
     end
+
+    describe '#expiry_time' do
+      before(:each) do
+        subject.expiry_time_hh = '11'
+        subject.expiry_time_mm = '11'
+        subject.expiry_time_meridian = 'am'
+      end
+
+      it 'displays error if all fields are blank' do
+        subject.expiry_time_hh = nil
+        subject.expiry_time_mm = nil
+        subject.expiry_time_meridian = nil
+        subject.valid?
+        expect(subject.errors.messages[:expiry_time]).to eq(['can\'t be blank'])
+      end
+
+      validate_expiry_time_hours = [
+        { value: nil, errors: ['can\'t be blank'] },
+        { value: 'not a number', errors: ['is in wrong format'] },
+        { value: '14', errors: ['is in wrong format'] },
+        { value: '0', errors: ['is in wrong format'] },
+      ]
+
+      validate_expiry_time_hours.each do |h|
+        it "displays '#{h[:errors][0]}' error when hours field is #{h[:value]}" do
+          subject.expiry_time_hh = h[:value]
+          subject.valid?
+          expect(subject.errors.messages[:expiry_time]).to eq(h[:errors])
+        end
+      end
+
+      validate_expiry_time_minutes = [
+        { value: nil, errors: ['can\'t be blank'] },
+        { value: 'not a number', errors: ['is in wrong format'] },
+        { value: '-6', errors: ['is in wrong format'] },
+        { value: '66', errors: ['is in wrong format'] },
+      ]
+
+      validate_expiry_time_minutes.each do |m|
+        it "displays '#{m[:errors][0]}' error when hours field is #{m[:value]}" do
+          subject.expiry_time_mm = m[:value]
+          subject.valid?
+          expect(subject.errors.messages[:expiry_time]).to eq(m[:errors])
+        end
+      end
+
+      it 'displays error if am/pm field is blank' do
+        subject.expiry_time_meridian = ''
+        subject.valid?
+        expect(subject.errors.messages[:expiry_time]).to eq(['must be am or pm'])
+      end
+
+      it 'displays only one error message at a time' do
```

Comment:
I don't feel like the test description fits the test expectation here. 
Based on the description, shouldn't the expectation be a number?

`expect(subject.errors.messages[:expiry_time].count).to eq(1)`?

Or should the test description be changed?

Reply:
Yes, thanks for flagging this! Resolved this in e543ec7